### PR TITLE
feat(client): ✨ Add Skip as voting option

### LINKED ISSUE #134: Skip vote and reduce timer when everyone votes
Catniss should be able to click and "Abstain" button to skip the vote because maybe they want to spend more time gathering information during the game

The timer for voting should be reduced once everyone has entered their vote because having to wait the whole time would get annoying.

Task List
- [x] the type of uid is `string | null` on the vote in firestore for abstaining vote.
- [x] Make "Skip Vote" button in frontend.
- [x] Check if skip votes is >= any other option in backend.
- [x] Make timer shorter when all players have voted in backend.

Acceptance Criteria
- Timer goes to set value when everyone has either voted or abstained
- the abstain button works
- if more people obtained than voted for a particular player then no on is voted out
- hacker should not be able go to the reduced the timer before everyone has voted

### PR DESCRIPTION
**Pull request by njumaah**
close: #134

### CODE REVIEW

**Review by Muaad2001 — DISMISSED** [review]
looks good

**Review by mrjoshua520 — COMMENTED** [review]
Everything looks good to me so this is kinda an approval but I personally think the skip vote button being centered below the players grid would look better. I think we should ask the group before merging

**Review by mrjoshua520 — APPROVED** [review]

File: functions/src/lobby.ts [hunk 1]
```diff
@@ -439,20 +440,41 @@ export const onVoteWrite = functions.firestore.document("/lobbies/{code}/votes/{
       return;
     }
 
-    const { players, uids } = lobbyData;
+    const { players, uids, skipVote: oldSkipVote } = lobbyData;
+    let skipVote = oldSkipVote;
```
Comment (gchamb) [inline functions/src/lobby.ts:444]: why create an alias if you create another variable with the same name?

**Review by gchamb — DISMISSED** [review]
left a minor comment above. then ill merge it

File: functions/src/lobby.ts (hunk 1 above)
Comment (nstringham) [reply functions/src/lobby.ts:444]: I think this is done so that it is mutable but maybe it would be better to do a separate deconstruction with let

Comment (njumaah) [reply functions/src/lobby.ts:444]: i think its fine the way it is

**Review by gchamb — APPROVED** [review]
left a minor comment above. then ill merge it

**Pull request merged**